File: legacy_restore/src_local/agents/Support/silent_error_detector.py
"""
SISTEMA DE PERSONAS AGENTES - SUPORTE TÉCNICO
Módulo: Detector de Erros Silenciados (SilentErrorDetector)
Função: Especialista em identificar capturas de exceção sem tratamento via AST.
"""
import ast
import logging

logger = logging.getLogger(__name__)

class SilentErrorDetector:
    def __init__(self, judge):
        self.judge = judge

    def detect(self, tree, rel_path, lines, agent_name, ignore_test_context=False):
        """Identifica padrões de falha lógica via análise AST."""
        issues = []
        for node in ast.walk(tree):
            if self._is_silent_except(node, tree, ignore_test_context):
                issues.append(self._create_issue(node, rel_path, lines, agent_name))
        return issues

    def _is_silent_except(self, node, tree, ignore_test_context):
        if not isinstance(node, ast.ExceptHandler): return False
        
        # Se for bare exce-pt ou exce-pt: pa-ss/continue
        is_empty = not node.type or (len(node.body) == 1 and isinstance(node.body[0], (ast.Pass, ast.Continue)))
        if is_empty:
            if self.judge.is_node_safe(node, tree, ignore_test_context=ignore_test_context):
                return False
            return True
        return False

    def _create_issue(self, node, rel_path, lines, agent_name):
        logger.warning(f"Silent error capture detected in {rel_path}:{node.lineno}")
        i = node.lineno - 1
        return {
            'file': rel_path, 'line': node.lineno, 
            'issue': 'Captura de erro silenciosa detectada.',
            'severity': 'high', 'context': agent_name,
            'snippet': "\n".join(lines[max(0, i-2):min(len(lines), i+3)])
        }

File: src_local/agents/Support/Analysis/silent_error_detector.ts
/**
 * SISTEMA DE PERSONAS AGENTES - SUPORTE TÉCNICO
 * Módulo: Detector de Erros Silenciados (SilentErrorDetector)
 */
import winston from "winston";
import { HubManagerGRPC } from "../../../core/hub_manager_grpc.ts";

const logger = winston.child({ module: "SilentErrorDetector" });

export interface SilentErrorIssue {
    file: string; line: number; issue: string; severity: string; context: string; snippet: string;
}

/**
 * 🔇 SilentErrorDetector — Detecta padrões de falha lógica silenciosa.
 */
export class SilentErrorDetector {
    constructor(private hubManager?: HubManagerGRPC) {}

    async detect(content: string, filePath: string): Promise<SilentErrorIssue[]> {
        if (filePath.includes("silent_error_detector.ts")) return [];
        
        if (!this.hubManager) {
            logger.warn("⚠️ HubManager not provided to SilentErrorDetector, falling back to empty results.");
            return [];
        }

        const result = await this.hubManager.analyzeFile(filePath, content);
        if (!result || !result.findings) return [];

        const issues: SilentErrorIssue[] = result.findings
            .filter((f: any) => f.category === "SILENT_ERROR")
            .map((f: any) => ({
                file: filePath,
                line: f.line,
                issue: f.message,
                severity: f.severity === "CRITICAL" ? "HIGH" : "MEDIUM",
                context: "SilentErrorDetector (AST)",
                snippet: f.snippet || ""
            }));

        if (issues.length > 0) {
            logger.debug(`🔇 ${issues.length} silent errors detected via AST in: ${filePath}`);
        }

        return issues;
    }
}
